tests/algorithms/test_dagger.py

```diff
@@ -105,6 +105,47 @@ def get_random_acts(obs):
     assert nonzero_acts == 0
 
 
+def test_traj_collector_reproducible(tmpdir, pendulum_venv):
+    # Check that we get the same results if we run the collector
+    # twice with the same random seeds.
+    # In particular, check that the observations from all the
+    # collected trajectories are the same in each run.
+    trajs_obs_list = []
+    with torch.random.fork_rng():
+        for run_idx in range(2):
+            # Reset the random seeds.
+            save_dir = os.path.join(tmpdir, "run{0}".format(run_idx))
+            rng = np.random.default_rng(12345)
+            pendulum_venv.seed(12345)
+            pendulum_venv.action_space.seed(12345)
+
+            # Run the collector.
+            collector = dagger.InteractiveTrajectoryCollector(
+                venv=pendulum_venv,
+                get_robot_acts=lambda o: [
+                    pendulum_venv.action_space.sample() for _ in range(len(o))
+                ],
+                beta=0.5,
+                save_dir=save_dir,
+                rng=rng,
+            )
+            collector.reset()
+            zero_acts = np.zeros(
+                (pendulum_venv.num_envs,) + pendulum_venv.action_space.shape,
+                dtype=pendulum_venv.action_space.dtype,
+            )
+            for i in range(1000):
+                _, _, dones, _ = collector.step(zero_acts)
+
+            # Get the observations from all the collected trajectories.
+            file_paths = glob.glob(os.path.join(save_dir, "dagger-demo-*.npz"))
+            trajs = [types.load(p)[0] for p in file_paths]
+            trajs_obs = np.stack([trajs[i].obs for i in range(len(trajs))], axis=0)
+            trajs_obs_list.append(trajs_obs)
+
+    np.testing.assert_almost_equal(trajs_obs_list[0], trajs_obs_list[1])
```

Comment:
Does this fail if you do exact equality rather than almost equality?

If we're seeding everything the same I think we should be getting same results even up to floating point error. I don't particularly care if we don't, but provided exact equality holds I think we can compare the `Trajectory` objects directly rather than using the observations, which would be a stronger check (would make sure actions, rewards, etc are the same not just observations).